### PR TITLE
Add support for `eventtype` in Client.get_events()

### LINKED ISSUE #2778: Missing `eventtype` parameter in `fdsn.client.Client.get_events`
[Version 1.2 of the FDSN event web service](https://www.fdsn.org/webservices/fdsnws-event-1.2.pdf) (released on 2019-04-02) introduced the `eventtype` parameter.

Unless I am mistaken, this is not implemented in ObsPy.

Is someone working on that? Otherwise I can take a stab at it.

### PR DESCRIPTION
**Pull request by claudiodsf**
### What does this PR do?

[Version 1.2 of the FDSN event web service](https://www.fdsn.org/webservices/fdsnws-event-1.2.pdf) (released on 2019-04-02) introduced the `eventtype` parameter.
This is already implicitly supported by ObsPy, given the way the FDSN client is written.
~~It only needs documentation, which is the scope of this PR.~~
It actually needs some editing of the list of allowed parameters (see the commits) and some testing (TODO).

### Why was it initiated?  Any relevant Issues?

Should fix #2778.

### PR Checklist
- [X] Correct base branch selected? `master` for new features, `maintenance_...` for bug fixes
- [ ] This PR is not directly related to an existing issue (which has no PR yet).
- [X] If the PR is making changes to documentation, docs pages can be built automatically.
      Just remove the space in the following string after the + sign: "+DOCS"
- [x] If any network modules should be tested for the PR, add them as a comma separated list
      (e.g. `clients.fdsn,clients.arclink`) after the colon in the following magic string: "+TESTS:"
      (you can also add "ALL" to just simply run all tests across all modules)
- [ ] All tests still pass.
- [ ] Any new features or fixed regressions are be covered via new tests.
- [ ] Any new or changed features have are fully documented.
- [x] Significant changes have been added to `CHANGELOG.txt` .
- [ ] First time contributors have added your name to `CONTRIBUTORS.txt` .

### CODE REVIEW

**Comment by claudiodsf** [comment]
I would like to check the +DOCS build for this one, but it didn't seem to work...

**Comment by d-chambers** [comment]
Does `eventtype` need to be added to any of the lists in `obspy.clients.fdsn.header` for this to work? It might be good to figure out some way to test this without an explicit network call. I am not very familiar with this part of ObsPy but could you add a test that makes sure eventtype is included in the output of `obspy.clients.fdsn.client.setup_query_dict` to make sure it is eventually included in the generated url?

**Comment by claudiodsf** [comment]
Thanks @d-chambers, I didn't notice `clients.fdsn.header`, since it works even without updating that file (why? 🤔)
Anyway, I will update this PR with changes to `clients.fdsn.header`.

Not sure if this can be tested without network... I'll check this out.

Any idea why DOCS are not built ?

File: obspy/clients/fdsn/client.py
Comment (claudiodsf) [inline obspy/clients/fdsn/client.py:483]: The reason why I would like to bud +DOCS is that I'm not sure whether this link works

**Comment by claudiodsf** [comment]
I tired to build the +DOCS locally, but `make html` fails retrieving the asset http://tests.obspy.org/snippets/navbar.html 

Not sure if @megies knows this already or if I need to create a new issue.

**Comment by barsch** [comment]
docs building is broken for python 3.x - I tried to fix it last year but ultimately failed due to time restrictions - see: https://github.com/obspy/obspy/pull/2662 - feel free to contribute ;)

anyway I created empty files "http://tests.obspy.org/snippets/navbar.html" and "http://tests.obspy.org/snippets/footer.html" so at least you can try to build locally - good luck ...

**Comment by claudiodsf** [comment]
Thanks @barsch! I'll try to build locally under Python 2.

**Comment by barsch** [comment]
@claudiodsf let me know if you need any other remote file

**Comment by claudiodsf** [comment]
Remote files are ok, but I'm still unable to build docs, even under py2...

It looks too complex to me, given what I want to do, so I'll probably give up... sorry.

My goal was to check whether the instruction  ```   :const:`obspy.core.event.header.EventType` ``` produces the right hyperlink in the documentation.

Maybe someone can just confirm or correct that? Otherwise I'll remove it.

**Comment by claudiodsf** [comment]
@d-chambers Not sure how to test this other than adding a network test in `test_client.py` with a call to `client.get_events()`.

But there are many parameters of `get_events()` that are untested! Is testing `eventtype` really necessary?

Also, to my knowledge, the only webservice to date which implements `fdsnws-event-1.2` is RESIF.

**Comment by d-chambers** [comment]
Ah I see. I guess if other tests pass it should be fine. The main thing is making sure existing network tests still work as well as they did before :smile:. I don't think adding another network test (which can be brittle and slow) is needed for such a small feature, especially if we haven't been testing each option in `get_events`. 

Feel free to merge it.

File: obspy/clients/fdsn/client.py
Comment (claudiodsf) [reply obspy/clients/fdsn/client.py:483]: I could build the docs locally thanks to the excellent work of by @heavelock https://github.com/heavelock/obspy/pull/9 (merci !) and verified that this works.

So, good to go!

**Comment by claudiodsf** [comment]
@d-chambers I cannot merge without an approving review. Would you mind doing it?

Also, do you want a Changelog entry for this?

**Comment by claudiodsf** [comment]
Let's do network tests just to be sure :wink: +TESTS:clients.fdsn

**Comment by claudiodsf** [comment]
The FDSN network test fails are unrelated to my code: those test fail also on current master due to extra spaces in the output.

So, from my standpoint, this is good to merge 😉

**Comment by claudiodsf** [comment]
Just, let me know if you want a Changelog entry.

**Review by d-chambers — APPROVED** [review]

**Comment by d-chambers** [comment]
> Just, let me know if you want a Changelog entry

A changelog entry is a good idea. After that feel free to merge (or I will if you can't).

**Comment by claudiodsf** [comment]
@d-chambers Done! But I cannot merge, so you'll have to 😉

**Comment by claudiodsf** [comment]
Thanks @d-chambers for taking care of this!

**Pull request merged**
